src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,540 @@
+    exact measure.le_add_right (le_refl _) },
+  { rw [singular_part, dif_neg h],
+    apply_instance }
+end
+
+instance {μ ν : measure α} [finite_measure μ] :
+  finite_measure (ν.with_density $ radon_nikodym_deriv μ ν) :=
+begin
+  by_cases h : have_lebesgue_decomposition μ ν,
+  { obtain ⟨-, -, h⟩ := have_lebesgue_decomposition_spec h,
+    refine finite_measure_le μ _,
+    conv_rhs { rw h },
+    exact measure.le_add_left (le_refl _) },
+  { rw [radon_nikodym_deriv, dif_neg h, with_density_zero],
+    apply_instance }
+end
+
+theorem eq_singular_part_of_have_lebesgue_decomposition
+  {μ ν : measure α} (h : have_lebesgue_decomposition μ ν)
+  (s : measure α) (f : α → ℝ≥0∞) (hs : s ⊥ₘ ν) (hadd : μ = s + ν.with_density f) :
+  s = μ.singular_part ν :=
+begin
+  obtain ⟨hmeas, hsing, hadd'⟩ := have_lebesgue_decomposition_spec h,
+  obtain ⟨⟨S, hS₁, hS₂, hS₃⟩, ⟨T, hT₁, hT₂, hT₃⟩⟩ := ⟨hs, hsing⟩,
+  rw hadd' at hadd,
+  have hνinter : ν (S ∩ T)ᶜ = 0,
+  { rw set.compl_inter,
+    refine nonpos_iff_eq_zero.1 (le_trans (measure_union_le _ _) _),
+    rw [hT₃, hS₃, add_zero],
+    exact le_refl _ },
+  have heq : s.restrict (S ∩ T)ᶜ = (μ.singular_part ν).restrict (S ∩ T)ᶜ,
+  { ext1 A hA,
+    have hf : ν.with_density f (A ∩ (S ∩ T)ᶜ) = 0,
+    { refine with_density_absolutely_continuous ν _ _,
+      rw ← nonpos_iff_eq_zero,
+      exact hνinter ▸ measure_mono (set.inter_subset_right _ _) },
+    have hrn : ν.with_density (μ.radon_nikodym_deriv ν) (A ∩ (S ∩ T)ᶜ) = 0,
+    { refine with_density_absolutely_continuous ν _ _,
+      rw ← nonpos_iff_eq_zero,
+      exact hνinter ▸ measure_mono (set.inter_subset_right _ _) },
+    rw [restrict_apply hA, restrict_apply hA, ← add_zero (s (A ∩ (S ∩ T)ᶜ)), ← hf,
+        ← add_apply, ← hadd, add_apply, hrn, add_zero] },
+  have heq' : ∀ A : set α, measurable_set A → s A = s.restrict (S ∩ T)ᶜ A,
+  { intros A hA,
+    have hsinter : s (A ∩ (S ∩ T)) = 0,
+    { rw ← nonpos_iff_eq_zero,
+      exact hS₂ ▸ measure_mono
+        (set.subset.trans (set.inter_subset_right _ _) (set.inter_subset_left _ _)) },
+    rw [restrict_apply hA, ← add_zero (s (A ∩ (S ∩ T)ᶜ)), ← hsinter, ← measure_union,
+        ← set.inter_union_distrib_left, set.compl_union_self, set.inter_univ],
+    { exact disjoint.inter_left' _ ( disjoint.inter_right' _ disjoint_compl_left) },
+    { measurability },
+    { measurability } },
+  ext1 A hA,
+  have hμinter : μ.singular_part ν (A ∩ (S ∩ T)) = 0,
+  { rw ← nonpos_iff_eq_zero,
+    exact hT₂ ▸ measure_mono
+      (set.subset.trans (set.inter_subset_right _ _) (set.inter_subset_right _ _)) },
+  rw [heq' A hA, heq, ← add_zero ((μ.singular_part ν).restrict (S ∩ T)ᶜ A), ← hμinter,
+      restrict_apply hA, ← measure_union, ← set.inter_union_distrib_left,
+      set.compl_union_self, set.inter_univ],
+  { exact disjoint.inter_left' _ ( disjoint.inter_right' _ disjoint_compl_left) },
+  { measurability },
+  { measurability }
+end
+
+theorem eq_radon_nikodym_deriv_of_have_lebesgue_decomposition
```

Comment:
can you also restate this by saying that `f = μ.radon_nikodym_deriv ν` almost everywhere, under the same assumptions (possibly assuming also that `f` is measurable if necessary)?

Reply:
Right, we will need a lemma saying `\mu.with_density f = \mu.with_density g -> f = g a.e.`. I'll make a separate PR for this since I think it might be a bit long

Reply:
ok, another PR seems like a good idea.